handler_test.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2021-2022 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package connect
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/bufbuild/connect-go/internal/assert"
+	pingv1 "github.com/bufbuild/connect-go/internal/gen/connect/ping/v1"
+)
+
+var errFailedToCreateStream = errors.New("failed to create stream")
+
+func TestInterceptorsCalledIfSenderReceiverNil(t *testing.T) {
+	t.Parallel()
+	mux := http.NewServeMux()
+	pingPath := "/connect.ping.v1.PingService/Ping"
+	handler := NewUnaryHandler[pingv1.PingRequest, pingv1.PingResponse](
```

Comment:
Have to use NewUnaryHandler here directly since the test needs to access the unexported `protocolHandlers` field so we can't use the connect generated code (import cycle).